### PR TITLE
Bump flow-go version to v0.15.4

### PR DESCRIPTION
**Pull request by jordanschalm**
## Description

Bumps `flow-go` dependency to `v0.15.4`, which includes latest breaking `fvm` changes refactoring errors. DKG integration testing in `flow-go` will use this version.
______

For contributor use:

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: transaction_test.go [hunk 1]
```diff
@@ -201,8 +201,8 @@ func TestSubmitTransaction_Invalid(t *testing.T) {
 		require.Error(t, result.Error)
 
 		assert.IsType(t, &types.FlowError{}, result.Error)
-		assert.IsType(t, &fvm.InvalidProposalKeySequenceNumberError{}, result.Error.(*types.FlowError).FlowError)
-		assert.Equal(t, invalidSequenceNumber, result.Error.(*types.FlowError).FlowError.(*fvm.InvalidProposalKeySequenceNumberError).ProvidedSeqNumber)
+		assert.IsType(t, &fvmerrors.InvalidProposalSeqNumberError{}, result.Error.(*types.FlowError).FlowError)
+		//assert.Equal(t, invalidSequenceNumber, result.Error.(*types.FlowError).FlowError.(*fvmerrors.InvalidProposalSeqNumberError).ProvidedSeqNumber)
```
Comment (jordanschalm) [inline transaction_test.go:205]: non-public field

**Review by janezpodhostnik — APPROVED** [review]

File: transaction_test.go (hunk 1 above)
Comment (ramtinms) [reply transaction_test.go:205]: yeah, I'm going to expose them as getter methods on an upcoming pr

**Review by ramtinms — APPROVED** [review]

File: transaction_test.go (hunk 1 above)
Comment (turbolent) [reply transaction_test.go:205]: @jordanschalm Can you please add a TODO with an explanation for why this is commented out?

Comment (jordanschalm) [inline transaction_test.go:205]:
```suggestion
		assert.IsType(t, &fvmerrors.InvalidProposalSeqNumberError{}, result.Error.(*types.FlowError).FlowError)
		// TODO ProvidedSeqNumber was changed to private field, un-comment following line when getter is added
		// https://github.com/onflow/flow-emulator/pull/41/files#r614266443
		//assert.Equal(t, invalidSequenceNumber, result.Error.(*types.FlowError).FlowError.(*fvmerrors.InvalidProposalSeqNumberError).ProvidedSeqNumber)
```

Comment (jordanschalm) [reply transaction_test.go:207]: https://github.com/onflow/flow-emulator/pull/41/commits/598df7abce4c17e4a0beff7c566df9e3b7993857

Comment (janezpodhostnik) [reply transaction_test.go:207]: The pr to expose the field went into flow-go yesterday: https://github.com/onflow/flow-go/pull/618

And I got this fixed in a emulator [PR](https://github.com/onflow/flow-emulator/pull/43) here https://github.com/onflow/flow-emulator/pull/43/files#diff-fa3545436305a9d76b9af6bfe8d963d15c8f223bebfecf7a82f353eff5e792daR215

**Pull request merged**